modules/service/variables.tf

```diff
@@ -0,0 +1,638 @@
+}
+
+variable "tasks_iam_role_use_name_prefix" {
+  description = "Determines whether the IAM role name (`tasks_iam_role_name`) is used as a prefix"
+  type        = bool
+  default     = true
+}
+
+variable "tasks_iam_role_path" {
+  description = "IAM role path"
+  type        = string
+  default     = null
+}
+
+variable "tasks_iam_role_description" {
+  description = "Description of the role"
+  type        = string
+  default     = null
+}
+
+variable "tasks_iam_role_permissions_boundary" {
+  description = "ARN of the policy that is used to set the permissions boundary for the IAM role"
+  type        = string
+  default     = null
+}
+
+variable "tasks_iam_role_tags" {
+  description = "A map of additional tags to add to the IAM role created"
+  type        = map(string)
+  default     = {}
+}
+
+variable "tasks_iam_role_policies" {
+  description = "Map of IAM role policy ARNs to attach to the IAM role"
+  type        = map(string)
+  default     = {}
+}
+
+variable "tasks_iam_role_statements" {
+  description = "A map of IAM policy [statements](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#statement) for custom permission usage"
+  type        = any
+  default     = {}
+}
+
+################################################################################
+# Task Set
+################################################################################
+
+variable "external_id" {
+  description = "The external ID associated with the task set"
+  type        = string
+  default     = null
+}
+
+variable "scale" {
+  description = "A floating-point percentage of the desired number of tasks to place and keep running in the task set"
+  type        = any
+  default     = {}
+}
+
+variable "force_delete" {
+  description = "Whether to allow deleting the task set without waiting for scaling down to 0"
+  type        = bool
+  default     = null
+}
+
+variable "wait_until_stable" {
+  description = "Whether terraform should wait until the task set has reached `STEADY_STATE`"
+  type        = bool
+  default     = null
+}
+
+variable "wait_until_stable_timeout" {
+  description = "Wait timeout for task set to reach `STEADY_STATE`. Valid time units include `ns`, `us` (or µs), `ms`, `s`, `m`, and `h`. Default `10m`"
+  type        = number
+  default     = null
+}
+
+################################################################################
+# Autoscaling
+################################################################################
+
+variable "enable_autoscaling" {
+  description = "Determines whether to enable autoscaling for the service"
+  type        = bool
+  default     = true
+}
+
+variable "autoscaling_min_capacity" {
+  description = "Minimum number of tasks to run in your service"
+  type        = number
+  default     = 1
+}
+
+variable "autoscaling_max_capacity" {
+  description = "Maximum number of tasks to run in your service"
+  type        = number
+  default     = 10
```

Comment:
```suggestion
  default     = null
```

Reply:
See [here](https://github.com/terraform-aws-modules/terraform-aws-ecs/pull/76#discussion_r1155118267) - autoscaling is enabled by default since `desired_count` is always ignored